_drafts/mejorando-nuestra-vida-con-activeadmin-addons.md

```diff
@@ -0,0 +1,196 @@
+
+## Uso
+
+A continuación veremos como implementar algunas de las mejoras más interesantes.
+
+### Select2
+
+![activeadmin_addons_2a](/images/activeadmin_addons/activeadmin_addons_2a.png)
+
+
+
+![activeadmin_addons_2b](/images/activeadmin_addons/activeadmin_addons_2b.png)
+
+Una vez instalada la gema todos los dropdown comenzarán a utilizar Select2 automáticamente lo que por defecto nos permite buscar dentro de los datos locales.
+
+#### Búsqueda por Ajax
+
+Para sacar los datos de otra parte se puede hacer lo siguiente:
+
+```ruby
+# Sin ajax
+f.input :category_id, as: :select, collection: Category.all {|category| [category.name, category.id] }
+
+# Con ajax
+f.input :category_id, as: :search_select, url: admin_categories_path, fields: [:name]
+```
+
+Para el campo `url` estamos usando un helper de ActiveAdmin que automáticamente nos da la url para el index de las categorías y como Select2 pide un JSON eso es lo que le entrega.
+
+`fields` representa qué campos en el json se van a buscar, por defecto es `:name` pero si el recurso no tiene esa columna el proceso fallará. Lo mismo ocurrirá con el campo opcional `display_name` que es lo que se muestra en el select.
+
+#### Tags (Selección múltiple)
+
+ ![select2-tags](/images/activeadmin_addons/select2-tags.gif)
+
+```ruby
+f.input :number, as: :tags, collection: ["1", "2"]
+```
+
+Ojo, no es lo mismo que `:has_many`, no crea relaciones entre tablas sino que produce una string concadenada con `,`.
+
+
+
+#### Dropdowns anidados
+
+A la hora de filtrar datos ActiveAdmin Addons trae la posibilidad de anidar dropdowns, haciendo dependiente la carga de resultados de una selección anterior.
+
+ ![select2-nested-select-default](/images/activeadmin_addons/select2-nested-select-default.gif)
+
+```ruby
+f.input :city_id, as: :nested_select, fields: [:name],
+                  level_1: {
+                    attribute: :country_id
+                  },
+                  level_2: {
+                    attribute: :region_id
+                  },
+                  level_3: {
+                    attribute: :city_id,
+                    minimum_input_length: 1,
+                    fields: [:name, :information]
+                  }
+```
+
+El código anterior automáticamente carga los resultados del `index` de ActiveAdmin de cada recurso, al igual como lo hacía `admin_categories_path` más arriba.
+
+
+
+### Etiquetas
+
+ActiveAdmin trae _status tags_ (  ![tag](/images/activeadmin_addons/tag.png)) que por defecto se pueden usar de la siguiente manera con _enum_:
+
+```ruby
+column :status do |d|
+  status_color = {
+    closed: 'ok',
+    pending: 'yes',
+    created: 'error'
+    }
+  status_tag(d.status, status_color[d.status.to_sym])
+end
+```
+
+En vez de estar creando un bloque por cada etiqueta, ActiveAdmin Addons resume la funcionalidad en una sola función:
+
+```ruby
+tag_column :status
```

Comment:
@blackjid @gmq me parece que no están hablando de lo mismo. Yo le hubiese contestado al Negro que `tag_column/row` funciona con las gemas [Enumerize](https://github.com/brainspec/enumerize) o [RailsEnum](http://api.rubyonrails.org/v4.1/classes/ActiveRecord/Enum.html) que viene incluido en Rails como se puede ver en la [implementación](https://github.com/platanus/activeadmin_addons/blob/master/lib/activeadmin_addons/addons/enum_tag.rb#L5). Por lo que el ejemplo usando:
```
status_color = {
  closed: 'ok',
  pending: 'yes',
  created: 'error'
}
```
no debería funcionar (ya que lanza la excepción `"need to pass an enumerize or enum attribute"`) y se debería definir los enum usando alguna de esas gemas.

Si entiendo bien lo que dice Memo, yo respondería: es verdad! los colores de cada tag se definen en el css en base al nombre de cada uno y no, no es posible pasar otras clases.